src/mesh_tools.rs

```diff
@@ -70,6 +214,63 @@ mod tests {
         Vector3::new(x, y, z)
     }
 
+    fn tessellated_triangle_geometry_after_welding() -> Geometry {
```

Comment:
This is dependent on the welding parameter. Not sure I am a fan of creating these data functions that are only used once. Shouldn't this be inlined in the tests? Also below.

Reply:
I'm not entirely against this. I think it's okay since it gives a name to something complicated, it doesn't pollute code of the test case and you can focus on more important code when looking at it.